_stbt/stbt_run.py

```diff
@@ -53,12 +53,29 @@ def video(args, dut):
 
 
 def _import_by_filename(filename_):
-    module_dir, module_file = os.path.split(filename_)
+    from importlib import import_module
+    module_dir, module_file = os.path.split(os.path.abspath(filename_))
     module_name, module_ext = os.path.splitext(module_file)
     if module_ext != '.py':
         raise ImportError("Invalid module filename '%s'" % filename_)
-    sys.path = [os.path.abspath(module_dir)] + sys.path
-    return __import__(module_name)
+
+    import_dir = module_dir
+    import_name = module_name
+    while os.path.exists(os.path.join(import_dir, "__init__.py")):
+        import_dir, s = os.path.split(import_dir)
+        import_name = "%s.%s" % (s, module_name)
```

Comment:
I don't see how this will work for multiple levels of import dir. I've added a test in my fixup commit.

Reply:
Fixed